### ISSUE TITLE
Concerto 1.0 generates identifiers even when they should be user-provided

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/concerto#244]
### Description

I think this should be a bug? I thought user-declared identifiers (with `identified by XXX`) were now meant to be provided by the user -- and distinguished from system generated identifiers (with `identified`).

But Concerto `1.0.0-alpha.2` generates user-declared identifiers. For instance:

**Model**
```
namespace org.test

concept C identified by id {
	o String name
	o String id
}
```

**Data**
```
{
    "$class": "org.test.C",
    "name": "John"
}
```

Validates as:
```
zsh-5.8$ concerto validate --ctoFiles test6.cto --sample test6.json  
10:49:53 AM - INFO: 
{
  "$class": "org.test.C",
  "name": "John",
  "id": "e55698df-2218-40ee-bcf9-13e6bfb38958"
}
```

**Comment by jeromesimeon** [comment]
I'm unclear if that is the intended behaviour though.

**Comment by dselman** [comment]
That's a bug imo - `id` is declared as an identifier and identifiers are never optional.

**Comment by jeromesimeon** [comment]
> That's a bug imo - `id` is declared as an identifier and identifiers are never optional.

That makes sense to me too!

I think the issue here is that `id` is automatically generated (so it behaves is if it had a default generated value, even when not optional).

**Comment by jeromesimeon** [comment]
Addressed in #246:

```
zsh-5.8$ more test.cto
namespace org.test

concept C identified by id {
        o String name
        o String id
}
zsh-5.8$ more test.json
{
    "$class": "org.test.C",
    "name": "John"
}
zsh-5.8$ ./index.js validate --model test.cto --input test.json
2:16:00 PM - INFO: Input is invalid
2:16:00 PM - ERROR: Invalid or missing identifier for Type C in namespace org.test
zsh-5.8$ ./index.js validate --model test.cto --input test.json --functional
2:16:01 PM - INFO: Input is invalid
2:16:01 PM - ERROR: Instance org.test.C#undefined has an empty identifier.
```
